kube-client/src/config/file_config.rs

```diff
@@ -574,14 +593,28 @@ users:
     client-certificate-data: aGVsbG8K
     client-key-data: aGVsbG8K
 "#;
-        let file = tempfile::NamedTempFile::new().expect("create config tempfile");
-        fs::write(file.path(), config_yaml).unwrap();
-        let cfg = Kubeconfig::read_from(file.path())?;
+        let cfg = Kubeconfig::from_text(config_yaml)?;
 
         // Ensure we have data from both documents:
         assert_eq!(cfg.clusters[0].name, "k3d-promstack");
         assert_eq!(cfg.clusters[1].name, "k3d-k3s-default");
 
         Ok(())
     }
+
+    #[test]
+    fn kubeconfig_from_empty_string() -> Result<(), KubeconfigError> {
+        let cfg = Kubeconfig::from_text("")?;
+
+        assert!(cfg.preferences.is_none());
+        assert!(cfg.clusters.is_empty());
+        assert!(cfg.auth_infos.is_empty());
+        assert!(cfg.contexts.is_empty());
+        assert!(cfg.current_context.is_none());
+        assert!(cfg.extensions.is_none());
+        assert!(cfg.kind.is_none());
+        assert!(cfg.api_version.is_none());
```

Comment:
this feels like it is testing `Kubeconfig::default() == Kubeconfig::from_text("")` (if we had derived `Default`).

think it's probably better to derive `Default`, test that, and additionally let people create fake `Kubeconfig` an easier way

Reply:
You are right, this is exactly that the reason - I'd be happy to add `impl Default for Kubeconfig`.

Reply:
Wait, a sec, not `Default`, but `PartialEq`. Correct?

Reply:
Done